packages/flutter_tools/lib/src/vmservice.dart

```diff
@@ -321,9 +326,10 @@ Future<FlutterVmService> _connect(
   PrintStructuredErrorLogMethod printStructuredErrorLogMethod,
   io.CompressionOptions compression = io.CompressionOptions.compressionDefault,
   Device device,
+  @required Logger logger,
 }) async {
-  final Uri wsUri = httpUri.replace(scheme: 'ws', path: globals.fs.path.join(httpUri.path, 'ws'));
```

Comment:
I think this is OK, because by construction the path is always going to end in a `/` and we were really relying on a quirk of path.join on windows to make this work.

The correct fix would be to expose the url style of package:path as a top level

Reply:
actually fixing this one in https://github.com/flutter/flutter/pull/79566